### DISCUSSION TITLE
Can't put floating option to the right.

### DISCUSSION THREAD

**Discussion by fnacarellidev** [nvim-tree/nvim-tree.lua#1765] [Q&A] [answered]
Here's my current `nvim-tree.lua` config file
`
require("nvim-tree").setup({
	view = {
		side = "right",
		float = {
			enable = true,
			quit_on_focus_loss = false,
		},
		mappings = {
			list = {
				{ key = "u", action = "dir_up" },
				{ key = "t", action = "cd" },
			},
		},
	},
	renderer = {
		highlight_opened_files = "all",
		indent_markers = {
			enable = true,
		},
	},
})
`
I thought that if I did that the nvim-tree would float and would be at the right of my screen, but it doesn't, anyone knows why?

**Comment by alex-courtis** [comment, marked as answer]
```lua
require("nvim-tree").setup {
  view = {
    side = "right",
    float = {
      enable = true,
      quit_on_focus_loss = false,
    },
  },
}
```

`view.side` does not apply to floating windows. You can set all your floating window parameters via `view.float`.

See `:help nvim_open_win` for all the options. `anchor` might be the one you are looking for.

**Reply by fnacarellidev** [reply]
Thank you, do you have any advice on how to do this configuration for nvim-tree?

**Reply by alex-courtis** [reply]
Unfortunately I'm not familiar with nvim_open_win

If you can't achieve the behaviour you want with a static table, you can set `open_win_config` to a function which can dynamically calculate the `config`.

**Comment by lleger** [comment]
@fnacarellidev Did you manage to figure this out? It seems when I do the following the anchor doesn't change (or perhaps I'm misunderstanding the option). I am also trying to move the floating window to the right.

```
float = {
  enable = true,
  open_win_config = {
    anchor = "NE"
  }
}
```

**Reply by alex-courtis** [reply]
Take a look a @kryzar 's excellent solution:

https://github.com/nvim-tree/nvim-tree.lua/pull/1538#issuecomment-1336148668

That allows exact positioning of the floating window.

Hopefully @kryzar will add his [Recipe](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes) to the wiki ;)

**Reply by lleger** [reply]
This looks great! But is a bit more involved, was hoping anchor would just fix it to the right. Is there a reason that attribute doesn't seem to work?

**Reply by alex-courtis** [reply]
> This looks great! But is a bit more involved, was hoping anchor would just fix it to the right. Is there a reason that attribute doesn't seem to work?

Unfortunately that's outside the scope of nvim-tree, as the options are simply passed through. It's also very far outside of my knowledge.

`:help nvim_open_win`

**Reply by kryzar** [reply]
I really shouldn't be credited for this! I will post the recipe, but all I did was using [@davidsierradz's solution](https://github.com/nvim-tree/nvim-tree.lua/pull/1538#issuecomment-1223314278).

**Reply by jonoisart** [reply]
I modified @kryzar solution to pin the floating window to the upper right

``` lua
local WIDTH_RATIO = 0.3
local HEIGHT_RATIO = 0.8
local OFFSET = 3

require('nvim-tree').setup({
  view = {
    float = {
      enable = true,
      open_win_config = function()
        local screen_w = vim.opt.columns:get()
        local screen_h = vim.opt.lines:get() - vim.opt.cmdheight:get()
        local window_w = screen_w * WIDTH_RATIO
        local window_h = screen_h * HEIGHT_RATIO
        local window_w_int = math.floor(window_w)
        local window_h_int = math.floor(window_h)

        -- adjust for the offset
        local col_right_aligned = screen_w - window_w_int - OFFSET
        local row_offset = OFFSET - 3


        return {
          border = 'rounded',
          relative = 'editor',
          row = row_offset,
          col = col_right_aligned,
          width = window_w_int,
          height = window_h_int,
        }
      end,
    },
    width = function()
      return math.floor(vim.opt.columns:get() * WIDTH_RATIO)
    end,
  },
})
```

**Comment by kryzar** [comment]
Recipe is [here](https://github.com/nvim-tree/nvim-tree.lua/wiki/Recipes#center-a-floating-nvim-tree-window).

**Reply by alex-courtis** [reply]
Many thanks!
